### ISSUE TITLE
[DeviceConfiguration] logical interface names

### ISSUE THREAD

**Issue by nemesisdesign** [interop-dev/netjson#32]
Some OpenWRT developers suggested to add a distinction between physical interface names and logical names. This because physical names can be automatically generated and vary a lot from manufacturer to manufacturer.

I think the current `name` attribute of interface objects could refer to logical names while an optional `device` attribute may refer to the physical name of the interface.

**Comment by HRogge** [comment]
I am not sure how many routing protocols even KNOW the logical name. Olsrd does not... Olsrd2 does not.

**Comment by nemesisdesign** [comment]
@HRogge ok, that's true, but why should a routing protocol be concerned about logical interface names in a ``DeviceConfiguration`` object?

``DeviceConfiguration`` already has a similar distinction in the ``radios`` section: ``name`` refers to a logical radio name, while ``phy`` refers to a hardware setting.

Let's not forget that the primary goal of ``DeviceConfiguration`` is to configure a device.

**Comment by HRogge** [comment]
Sorry, didn't saw that it was ONLY for Device Configuration... because interfaces are also interesting for the Network Routes (outgoing interface).

"Logical Interfaces" are something quite restricted to OpenWRT, not sure other Operation Systems use them.

**Comment by nemesisdesign** [comment]
True, logical names are very popular in OpenWRT, I haven't been able to find a lot of information on other systems.

The predictability of interface names has been a problem and is still a problem for many OS, i've found more about it here:

- https://en.wikipedia.org/wiki/Consistent_Network_Device_Naming
- http://digitizor.com/2011/01/25/fedora-15-network-device-naming/
